### PR TITLE
IDE settings and contributing documentation

### PR DESCRIPTION
**Pull request by soleneweill**
## Description and Context
Default CONTRIBUTING.md and VScode settings. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] No code (non-breaking change that does not impact code: formatting, build system, documentation...)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Release (version number increment, possibly with documentation updates)

### CODE REVIEW

File: VSCODE_SETTINGS/settings.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  // global settings
+  "editor.codeActionsOnSave": {
+    "source.organizeImports": "explicit",
+    "source.formatDocument": "explicit",
+    "source.lint": "explicit",
+    "source.fixAll.eslint": "explicit"
+  },
+  "editor.formatOnPaste": false,
+
+  // backend settings, require vscode extensions: black, flake8, isort
```
Comment (EpigeneMax) [inline VSCODE_SETTINGS/settings.json]: I thought we wanted to leverage `ruff` in VSCode too.

Comment (soleneweill) [reply VSCODE_SETTINGS/settings.json]: Ok, can we be more precise on how we want to use it please? Still with the settings.json or with a ruff.toml file ? 
I am quite confuse on some parts of the file. I don't know how to replace it.

File: VSCODE_SETTINGS/ruff.toml
Comment (clemencepetit27) [inline VSCODE_SETTINGS/ruff.toml:1]: Do we want to enforce the docstring convention to numpy style in this file as well ?
Seems feasible according to the documentation : https://docs.astral.sh/ruff/faq/#does-ruff-support-numpy-or-google-style-docstrings

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+Please refer to the [Smart Commits documentation].(https://support.atlassian.com/bitbucket-cloud/docs/use-smart-commits/) 
+
+## Set VS CODE configuration
+
+To maintain code quality and ease developers contribution, we provide a minimal vscode settings.
+
+In each new repository, please make sure to include these settings by:
+* creating a `.vscode` folder
+* Adding the `settings.json` file to it. The file is accessible in the VSCODE_SETTINGS.
```
Comment (clemencepetit27) [inline CONTRIBUTING.md]: We should also add the `ruff.toml` file, right ?
Also, I think it's worth mentioning that the ruff extension should be installed on VScode.

**Review by clemencepetit27 — COMMENTED** [review]
Thanks @soleneweill for this PR ! I just left a couple of suggestions.

**Review by benjamin-millot — APPROVED** [review]

**Review by clemencepetit27 — APPROVED** [review]
Looks good to me, thanks !

**Review by benjamin-millot — APPROVED** [review]
Same, looking good!

**Pull request merged**